### DISCUSSION TITLE
failed to activate virtual environment

### DISCUSSION THREAD

**Discussion by neurodavidus** [wehr-lab/autopilot#73] [Q&A] [answered]
Hi again
Just for fun, I tried to install Autopilot on the terminal computer.
Does not seem to work.  I don't see any (autopilot) before the command line. I followed the instruction. Probably something silly but because deep down I don't understand how virtual environment work I am stick. 
A screen capture below:

![image](https://user-images.githubusercontent.com/4154936/107549047-a5438500-6bcf-11eb-9837-d2c3c18bf8a0.png)

thanks for your help. David

**Comment by mikewehr** [comment, marked as answer]
Hi David
I'm also still trying to figure out virtual environments.
maybe try something like this
source ~/.venv/autopilot/bin/activate

if that doesn't work, try to find where 'activate' is in the .venv directory and execute it
-Mike

**Reply by sneakers-the-rat** [reply]
this is correct, that was a mistake in the docs. it's been corrected here: https://github.com/wehr-lab/autopilot/commit/e82dfc27ba3bafad72c3a5bd5d3067442a072dfc

**Reply by neurodavidus** [reply]
ok thanks, it works now

**Comment by sneakers-the-rat** [comment]
Thanks for pointing out this problem with the docs! Virtualenvs can feel very mysterious, so the confusion is understandable. One surprisingly direct way to see how they work is to just look at what's in the `activate` file, for example by `cat /path/to/some/venv/bin/activate`. Here's one on my machine, abbreviated for clarity. Most of what it does is prepend the virtualenv to the `PATH` variable so the shell looks in the venv first, and then adds the name of the venv to the `PS1` variable that describes what text is printed to the left of your cursor in the shell (you can try changing it yourself just for funzies like `export PS1="very_fast_computer\$"`). hope that helps :)

```bash
VIRTUAL_ENV='/path/to/some/venv'
export VIRTUAL_ENV

_OLD_VIRTUAL_PATH="$PATH"
PATH="$VIRTUAL_ENV/bin:$PATH"
export PATH

# unset PYTHONHOME if set
if ! [ -z "${PYTHONHOME+_}" ] ; then
    _OLD_VIRTUAL_PYTHONHOME="$PYTHONHOME"
    unset PYTHONHOME
fi

if [ -z "${VIRTUAL_ENV_DISABLE_PROMPT-}" ] ; then
    _OLD_VIRTUAL_PS1="${PS1-}"
    if [ "x" != x ] ; then
        PS1="${PS1-}"
    else
        PS1="(`basename \"$VIRTUAL_ENV\"`) ${PS1-}"
    fi
    export PS1
fi

# Make sure to unalias pydoc if it's already there
alias pydoc 2>/dev/null >/dev/null && unalias pydoc || true

pydoc () {
    python -m pydoc "$@"
}

# This should detect bash and zsh, which have a hash command that must
# be called to get it to forget past commands.  Without forgetting
# past commands the $PATH changes we made may not be respected
if [ -n "${BASH-}" ] || [ -n "${ZSH_VERSION-}" ] ; then
    hash -r 2>/dev/null
fi


```

**Reply by neurodavidus** [reply]
It does! thanks
